File: frontend/src/types/operations.ts
export type BriefingType = "daily" | "weekly";
export type AutomationCadence = "daily" | "weekly";

export interface BriefingContent {
  headline: string;
  focus: string;
  nextTask: string;
  bottleneck: string;
  healthNarrative: string;
  actionPlan: string[];
  projectCount: number;
  riskCount: number;
  blockedTaskCount: number;
  teamHeadline: string;
}

export interface BriefingSnapshot {
  id: string;
  briefingType: BriefingType;
  periodKey: string;
  provider: string;
  content: BriefingContent;
  createdAt: string;
  updatedAt: string;
}

export interface AutomationRule {
  id: string;
  name: string;
  briefingType: BriefingType;
  cadence: AutomationCadence;
  enabled: boolean;
  maxRunsPerPeriod: number;
  lastRunAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface AutomationPreview {
  ruleId: string;
  periodKey: string;
  due: boolean;
  wouldCreateSnapshot: boolean;
  reason: string;
  actionSummary: string;
}

export interface AutomationRun {
  id: string;
  ruleId: string;
  mode: "execute";
  status: "completed" | "skipped";
  periodKey: string;
  outputSnapshotId?: string;
  decision: Record<string, unknown>;
  createdAt: string;
}

export interface AIProviderStatus {
  provider: "local" | "openai" | "anthropic" | "google";
  label: string;
  configured: boolean;
  enabled: boolean;
  executionMode: "active" | "disabled";
  detail: string;
}

export interface OperationsOverview {
  snapshots: BriefingSnapshot[];
  rules: AutomationRule[];
  recentRuns: AutomationRun[];
  providers: AIProviderStatus[];
}

File: backend/app/schemas/operations.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field, model_validator

BriefingType = Literal["daily", "weekly"]
AutomationCadence = Literal["daily", "weekly"]


class BriefingContent(BaseModel):
    headline: str
    focus: str
    next_task: str
    bottleneck: str
    health_narrative: str
    action_plan: list[str]
    project_count: int
    risk_count: int
    blocked_task_count: int
    team_headline: str


class BriefingSnapshotRead(BaseModel):
    id: str
    briefing_type: BriefingType
    period_key: str
    provider: str
    content: BriefingContent
    created_at: datetime
    updated_at: datetime


class BriefingGenerateRequest(BaseModel):
    briefing_type: BriefingType
    refresh_existing: bool = False


class AutomationRuleCreate(BaseModel):
    name: str = Field(min_length=3, max_length=120)
    briefing_type: BriefingType
    cadence: AutomationCadence

    @model_validator(mode="after")
    def cadence_matches_output(self) -> "AutomationRuleCreate":
        if self.briefing_type != self.cadence:
            raise ValueError("Briefing type and cadence must match")
        return self


class AutomationRuleUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=3, max_length=120)
    enabled: bool | None = None


class AutomationRuleRead(BaseModel):
    id: str
    name: str
    briefing_type: BriefingType
    cadence: AutomationCadence
    enabled: bool
    max_runs_per_period: int
    last_run_at: datetime | None
    created_at: datetime
    updated_at: datetime


class AutomationPreviewRead(BaseModel):
    rule_id: str
    period_key: str
    due: bool
    would_create_snapshot: bool
    reason: str
    action_summary: str


class AutomationExecuteRequest(BaseModel):
    confirm: bool


class AutomationRunRead(BaseModel):
    id: str
    rule_id: str
    mode: Literal["execute"]
    status: Literal["completed", "skipped"]
    period_key: str
    output_snapshot_id: str | None
    decision: dict
    created_at: datetime


class AIProviderStatus(BaseModel):
    provider: Literal["local", "openai", "anthropic", "google"]
    label: str
    configured: bool
    enabled: bool
    execution_mode: Literal["active", "disabled"]
    detail: str


class OperationsOverview(BaseModel):
    snapshots: list[BriefingSnapshotRead]
    rules: list[AutomationRuleRead]
    recent_runs: list[AutomationRunRead]
    providers: list[AIProviderStatus]
